plugins/airtable/src/SelectDataSource.tsx

```diff
@@ -39,12 +49,22 @@ export function SelectDataSource({ collection, onSelectDataSource }: SelectDataS
             setSelectedTableId("")
 
             const task = async () => {
-                const tables = await getTables(selectedBaseId, abortController.signal)
-                if (abortController.signal.aborted) return
-
-                setTables(tables)
-                setStatus("ready")
-                setSelectedTableId(tables[0]?.id ?? "")
+                try {
+                    const tables = await getTables(selectedBaseId, abortController.signal)
+                    if (abortController.signal.aborted) return
+
+                    setTables(tables)
+                    setStatus("ready")
+                    setSelectedTableId(tables[0]?.id ?? "")
+                } catch (error) {
+                    console.error(error)
+                    setStatus("error-tables")
+
+                    const baseName = selectedBase?.name ?? selectedBaseId
+                    framer.notify(`Failed to load tables for base "${baseName}". Check the logs for more details.`, {
```

Comment:
I’m not sure about the user experience here. We shouldn’t ask users to check their browser log for details. 

Instead, we could cover some common cases by inspecting the error message (such as a network issue) and for all other cases, display the error message prominently in the plugin window. Since such an error prevents the plugin from working, taking over the entire plugin window with the details could work.

This is how it looks like in Framer:

<img width="542" height="484" alt="Image" src="https://github.com/user-attachments/assets/6a029709-8a11-4ffd-a9d9-15a8fe09edbd" />

Reply:
I think error handling can be in a simple error component with a retry/recover functionality.